sdk/storage/storage-blob/review/storage-blob.api.ts

```diff
@@ -0,0 +1,1424 @@
+  customerProvidedKey?: Models.CpkInfo;
+  metadata?: Metadata;
+}
+
+// @public (undocumented)
+class BaseRequestPolicy implements RequestPolicy {
+  protected constructor(_nextPolicy: RequestPolicy, _options: RequestPolicyOptions);
+  // (undocumented)
+  readonly _nextPolicy: RequestPolicy;
+  // (undocumented)
+  readonly _options: RequestPolicyOptions;
+  log(logLevel: HttpPipelineLogLevel, message: string): void;
+  // (undocumented)
+  abstract sendRequest(webResource: WebResource): Promise<HttpOperationResponse>;
+  shouldLog(logLevel: HttpPipelineLogLevel): boolean;
+}
+
+// @public
+class BatchDeleteRequest extends BatchRequest {
+  constructor();
+  addSubRequest(url: string, credential: SharedKeyCredential | AnonymousCredential | TokenCredential, options?: BlobDeleteOptions): Promise<void>;
+}
+
+// @public
+class BatchRequest {
+  constructor();
+  // (undocumented)
+  protected addSubRequestInternal(subRequest: BatchSubRequest, assembleSubRequestFunc: () => Promise<void>): Promise<void>;
+  // (undocumented)
+  protected readonly batch: string;
+  // WARNING: The type "InnerBatchRequest" needs to be exported by the package (e.g. added to index.ts)
+  // (undocumented)
+  protected batchRequest: InnerBatchRequest;
+  getHttpRequestBody(): string;
+  getMultiPartContentType(): string;
+  getSubRequests(): Map<number, BatchSubRequest>;
+}
+
+// @public
+class BatchSetTierRequest extends BatchRequest {
+  constructor();
+  // WARNING: The type "Models.AccessTier" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.AccessTier" needs to be exported by the package (e.g. added to index.ts)
+  addSubRequest(url: string, credential: SharedKeyCredential | AnonymousCredential | TokenCredential, tier: Models.AccessTier, options?: BlobSetTierOptions): Promise<void>;
+}
+
+// @public (undocumented)
+interface BatchSubRequest {
+  credential: SharedKeyCredential | AnonymousCredential | TokenCredential;
+  url: string;
+}
+
+// @public (undocumented)
+interface BatchSubResponse {
+  _request: BatchSubRequest;
+  bodyAsText?: string;
+  errorCode?: string;
+  headers: HttpHeaders;
+  status: number;
+  statusMessage: string;
+}
+
+// @public
+interface BlobAbortCopyFromURLOptions {
+  abortSignal?: AbortSignalLike;
+  leaseAccessConditions?: Models.LeaseAccessConditions;
+}
+
+// @public
+interface BlobAcquireLeaseOptions {
+  abortSignal?: AbortSignalLike;
+  modifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+// @public
+interface BlobBreakLeaseOptions {
+  abortSignal?: AbortSignalLike;
+  modifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+// @public
+interface BlobChangeLeaseOptions {
+  abortSignal?: AbortSignalLike;
+  modifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+// @public
+class BlobClient extends StorageClient {
+  constructor(connectionString: string, containerName: string, blobName: string, options?: NewPipelineOptions);
```

Comment:
The set of overloads is different here. Can I not construct a BlobClient from a URL?

Reply:
Jeremy says this is the tool.